### PR TITLE
Add JS pagination for recent jobs in JobSeeker homepage

### PR DESCRIPTION
**Pull request by arun1595**
Fixes AgileVentures/MetPlus_tracker/issues/639

### CODE REVIEW

**Comment by patmbolger** [comment]
These aspects don't seem to work (?):

1. Sort by company name
2. Pagination

**Comment by arun1595** [comment]
@patmbolger I've fixed the issues mentioned. PTAL :smile:

**Comment by patmbolger** [comment]
I still don't see spring by company name working (?)

**Comment by arun1595** [comment]
@patmbolger Sorry, was in a hurry when working on it. 

The following error is thrown when I try to sort based on the company name:

```
ActionView::Template::Error (PG::AmbiguousColumn: ERROR:  column reference "created_at" is ambiguous
LINE 1: ... ON "companies"."id" = "jobs"."company_id" WHERE (created_at...
                                                             ^
: SELECT COUNT(DISTINCT "jobs"."id") FROM "jobs" LEFT OUTER JOIN "companies" ON "companies"."id" = "jobs"."company_id" WHERE (created_at > '2017-09-19 23:55:30.354264'))
```

I was able to fix it  by changing the `new_jobs` scope in `Job` model.

```ruby
scope :new_jobs, ->(given_time) { where('jobs.created_at > ?', given_time) }
```

If you know of a way to fix this issue, do let me know. Thanks!

**Comment by patmbolger** [comment]
This looks good.  Please update the PR so we can do a final review and merge.

**Comment by arun1595** [comment]
@patmbolger Done, PTAL :smile:

**Comment by patmbolger** [comment]
@arun1595 - you have some rspec test failures to be fixed.  See CI tests results.

File: spec/controllers/jobs_controller_spec.rb [hunk 1]
```diff
@@ -644,10 +619,7 @@
         expect(response).to render_template('jobs/_list_jobs')
       end
 
-      it 'recent_jobs: check_jobs' do
-        expect(assigns(:jobs).first.title).to eq 'Awesome new job 3'
       end
-    end
```
Comment (patmbolger) [inline spec/controllers/jobs_controller_spec.rb]: I don't want to lose these tests as they are useful.  You should be able to recover them with these changes:

1. jobs_controller_spec.rb - change line 553 to:
```
xhr :get, :list, job_type: 'my-company-all', page: 4
```
(the page name has been changed from `jobs_page` to just `page`)

2. jobs_controller.rb - change line 178 to:
```
@jobs = query.result.paginate(page: params[:page], per_page: items_per_page)
```
(make jobs an instance var so the tests can see it)

I'll be creating a PR this morning that also has these changes.

Comment (patmbolger) [reply spec/controllers/jobs_controller_spec.rb]: I merged in my PR this morning.  Rebasing against development should address my concerns for the controller tests I want to maintain.

Comment (arun1595) [reply spec/controllers/jobs_controller_spec.rb]: @patmbolger Thanks!

**Pull request merged**